book/src/user/experimental-features.md

```diff
@@ -0,0 +1,138 @@
+# Experimental features
+
+In `privacy-scaling-explorations/halo2` fork we have implemented many experimental features to cover different use cases, especially for those with huge circuits. We collect them in this page for easier tracking and referencing.
+
+## Commitment scheme abstraction
+
+To support different kind of polynomial commitment scheme, we added a trait `CommitmentScheme` to allow create/verify proof with different commitment scheme implementation, currently there are 2 available implementations in this fork:
+
+- `IPACommitmentScheme`
+
+  The original implementation from `zcash/halo2` with the original multi-open strategy `{Prover,Verifier}IPA`
+
+- `KZGCommitmentScheme`
+
+  KZG commitment scheme as in [GWC19](https://eprint.iacr.org/2019/953), which doesn't commit the instance columns, with 2 multi-open strategies available:
+
+  - `{Prover,Verifier}GWC` - The original strategy in [GWC19](https://eprint.iacr.org/2019/953)
+  - `{Prover,Verifier}SHPLONK` - The strategy proposed in [BDFG20](https://eprint.iacr.org/2020/081)
+
+When using `create_proof` and `verify_proof`, we need to specify the commitment scheme and multi-open strategy like:
+
+```rust
+// Using IPA
+create_proof<IPACommitmentScheme<_>, ProverIPA<_>, _, _, _, _>
+verify_proof<IPACommitmentScheme<_>, ProverIPA<_>, _, _, _>
+
+// Using KZG with GWC19 mutli-open strategy
+create_proof<KZGCommitmentScheme<_>, ProverGWC<_>, _, _, _, _>
+verify_proof<KZGCommitmentScheme<_>, ProverGWC<_>, _, _, _>
+
+// Using KZG with BDFG20 mutli-open strategy
+create_proof<KZGCommitmentScheme<_>, ProverSHPLONK<_>, _, _, _, _>
+verify_proof<KZGCommitmentScheme<_>, ProverSHPLONK<_>, _, _, _>
+```
+
+## `ConstraintSystem` extension
+
+### Dynamic lookup
+
+`ConstraintSystem::lookup_any` is added for use cases that need to lookup dynamic table instead of fixed table.
+
+Unlike `ConstraintSystem::lookup` which only allows `TableColumn`(s) as table, it allows any `Expression`(s) without simple selector.
```

Comment:
Would be nice to have an `examples` folder with one example for each one of these sections.

Reply:
Added in https://github.com/privacy-scaling-explorations/halo2/pull/226/commits/6909478a8292fe56777760c1e968c0865ef457b3